plugins/metrics/pkg/backend/metrics.go

```diff
@@ -408,3 +420,318 @@ func (m *MetricsBackend) UninstallCluster(ctx context.Context, in *emptypb.Empty
 	defer m.requestNodeSync(ctx, &corev1.Reference{})
 	return m.ClusterDriver.UninstallCluster(ctx, in)
 }
+
+// Metrics Remote Read Backend
+
+func targetAlreadyExistsError(id string) error {
+	return fmt.Errorf("target '%s' already exists", id)
+}
+
+func targetDoesNotExistError(id string) error {
+	return fmt.Errorf("target '%s' does not exist", id)
+}
+
+func getIdFromTargetMeta(meta *remoteread.TargetMeta) string {
+	return fmt.Sprintf("%s/%s", meta.ClusterId, meta.Name)
+}
+
+func (m *MetricsBackend) AddTarget(_ context.Context, request *remoteread.TargetAddRequest) (*emptypb.Empty, error) {
+	m.WaitForInit()
+
+	m.remoteReadTargetMu.Lock()
+	defer m.remoteReadTargetMu.Unlock()
+
+	targetId := getIdFromTargetMeta(request.Target.Meta)
+
+	if _, found := m.remoteReadTargets[targetId]; found {
+		return nil, targetAlreadyExistsError(targetId)
```

Comment:
Generally, errors returned from grpc handler functions should be of the form `status.Error(codes.XXX, ...)`, otherwise the grpc status code gets set to Unknown which makes it harder to handle errors on the client side.
```suggestion
		return nil, status.Errorf(codes.NotFound, "target not found: %s", id)
```